src/db/sqlite/models.rs

```diff
@@ -0,0 +1,1170 @@
+        Ok(result)
+    }
+
+    pub fn get_storage_timestamp_sync(&self, user_id: HawkIdentifier) -> Result<SyncTimestamp> {
+        let user_id = user_id.legacy_id as i64;
+        let modified = user_collections::table
+            .select(max(user_collections::modified))
+            .filter(user_collections::user_id.eq(user_id))
+            .first::<Option<i64>>(&self.conn)?
+            .unwrap_or_default();
+        Ok(SyncTimestamp::from_i64(modified)?)
+    }
+
+    pub fn get_collection_timestamp_sync(
+        &self,
+        params: params::GetCollectionTimestamp,
+    ) -> Result<SyncTimestamp> {
+        let user_id = params.user_id.legacy_id as u32;
+        let collection_id = self.get_collection_id(&params.collection)?;
+        if let Some(modified) = self
+            .session
+            .borrow()
+            .coll_modified_cache
+            .get(&(user_id, collection_id))
+        {
+            return Ok(*modified);
+        }
+        user_collections::table
+            .select(user_collections::modified)
+            .filter(user_collections::user_id.eq(user_id as i64))
+            .filter(user_collections::collection_id.eq(collection_id))
+            .first(&self.conn)
+            .optional()?
+            .ok_or_else(|| DbErrorKind::CollectionNotFound.into())
+    }
+
+    pub fn get_bso_timestamp_sync(&self, params: params::GetBsoTimestamp) -> Result<SyncTimestamp> {
+        let user_id = params.user_id.legacy_id as i64;
+        let collection_id = self.get_collection_id(&params.collection)?;
+        let modified = bso::table
+            .select(bso::modified)
+            .filter(bso::user_id.eq(user_id))
+            .filter(bso::collection_id.eq(&collection_id))
+            .filter(bso::id.eq(&params.id))
+            .first::<i64>(&self.conn)
+            .optional()?
+            .unwrap_or_default();
+        Ok(SyncTimestamp::from_i64(modified)?)
+    }
+
+    pub fn get_collection_timestamps_sync(
+        &self,
+        user_id: HawkIdentifier,
+    ) -> Result<results::GetCollectionTimestamps> {
+        let modifieds = sql_query(format!(
+            "SELECT {collection_id}, {modified}
+               FROM user_collections
+              WHERE {user_id} = ?
+               AND {collection_id} != ?",
+            collection_id = COLLECTION_ID,
+            user_id = USER_ID,
+            modified = LAST_MODIFIED
+        ))
+        .bind::<BigInt, _>(user_id.legacy_id as i64)
+        .bind::<Integer, _>(TOMBSTONE)
+        .load::<UserCollectionsResult>(&self.conn)?
+        .into_iter()
+        .map(|cr| SyncTimestamp::from_i64(cr.last_modified).map(|ts| (cr.collection, ts)))
+        .collect::<Result<HashMap<_, _>>>()?;
+        self.map_collection_names(modifieds)
+    }
+
+    fn check_sync(&self) -> Result<results::Check> {
+        // Check if querying works
+        sql_query("SELECT 1").execute(&self.conn)?;
+        Ok(true)
```

Comment:
sqlite does not support `SHOW STATUS` or similar queries. Since it seemed to just check if the database is reachable, I chose this way. But I'm not entirely sure.